facilitator/src/transport/s3.rs

```diff
@@ -136,23 +141,37 @@ impl S3Transport {
                         // the context of an async runtime when this callback is
                         // invoked, but because the closure is not declared
                         // async, we cannot use .await to work with Futures.
-                        let response = ureq::get(METADATA_SERVICE_TOKEN_URL)
-                            .query(
+
+                        let mut url = metadata_service_token_url();
+
+                        url.query_pairs_mut()
+                            .append_pair(
                                 "audience",
-                                format!("sts.amazonaws.com/{}", aws_account_id).as_ref(),
+                                &format!("sts.amazonaws.com/{}", aws_account_id),
                             )
-                            .set("Metadata-Flavor", "Google")
-                            // By default, ureq will wait forever to connect or
-                            // read.
-                            .timeout_connect(10_000) // ten seconds
-                            .timeout_read(10_000) // ten seconds
-                            .call();
-                        if response.error() {
-                            return Err(CredentialsError::new(format!(
+                            .finish();
+
+                        let mut request = prepare_request(
+                            None,
```

Comment:
nit: because there's no named parameters in Rust, it's not obvious what is being omitted here. Instead of having a single `prepare_request(Option<Agent>, RequestParameters)`, you could provide `prepare_request(RequestParameters)` and `prepare_request_with_agent(Agent, RequestParameters)`.

Reply:
I didn't even notice this because vscode was highlighting the labels for me: 

![image](https://user-images.githubusercontent.com/1465785/106325371-b47a1880-6248-11eb-8f85-828fe847b221.png)

I'll add that other function, but probably as `prepare_request_without_agent`, instead of `with`, considering majority of our usage of this function will come with the agent.